code/modules/mob/living/carbon/human/life.dm

```diff
@@ -1306,7 +1266,7 @@
 						if(2)	healths.icon_state = "health7"
 						else
 							//switch(health - halloss)
-							switch(100 - ((species && species.flags & NO_PAIN & !IS_SYNTHETIC) ? 0 : traumatic_shock))
+							switch(100 - traumatic_shock)

```

Comment:
IT doesn't exactly
NO_PAIN has been changed here.

Old behaviour: 
Species with no pain accumulate halloss as normal, but ignore all effects of it.

New Behaviour: 
Species with no pain don't accumulate halloss, except from certain species-specific mechanics. They suffer all the normal effects of halloss if it is applied to them

The idea here, think of halloss as a general system stress level rather than just pain. in normal organics it represents pain (and is applied by things that hurt). In diona, it represents something like sensory distress, and is only applied by darkness. Things that hurt don't apply halloss to them.

In future i'm thinking it could also be used for synthetics to represent fragmentation or system overload or somesuch, and be applied to them by EMP and similar specific anti-synthetic mechanics.

Basically, species with no_pain are still immune to pain. I've just reworked the mechanic to be a bit more flexible